test/e2e/tests/request-queuing/multiple-networks-dapps-txs.spec.js

```diff
@@ -0,0 +1,172 @@
+const {
+  withFixtures,
+  openDapp,
+  unlockWallet,
+  DAPP_URL,
+  DAPP_ONE_URL,
+  regularDelayMs,
+  WINDOW_TITLES,
+  defaultGanacheOptions,
+} = require('../../helpers');
+const { PAGES } = require('../../webdriver/driver');
+
+// TODO: Have to turn on the setting every time we want to test the setting!?!
+// TODO: Test this in prod, refresh the extension when the setting is on and you have to disable/enable it for the switchEthereumChain notification to work.
+
+describe('Request Queuing for Multiple Dapps and Txs on different networks.', function () {
+  it('should show switch network confirmations for per dapp selected networks when calling send transactions', async function () {
+    const port = 8546;
+    const chainId = 1338;
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerDoubleGanache()
+          .withPermissionControllerConnectedToTwoTestDapps()
+          .withSelectedNetworkControllerPerDomain()
+          .build(),
+        dappOptions: { numberOfDapps: 2 },
+        ganacheOptions: {
+          ...defaultGanacheOptions,
+          concurrent: {
+            port,
+            chainId,
+            ganacheOptions2: defaultGanacheOptions,
+          },
+        },
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        // Open account menu button
+        const accountOptionsMenuSelector =
+          '[data-testid="account-options-menu-button"]';
+        await driver.waitForSelector(accountOptionsMenuSelector);
+        await driver.clickElement(accountOptionsMenuSelector);
+
+        // Click settings from dropdown menu
+        const globalMenuSettingsSelector =
+          '[data-testid="global-menu-settings"]';
+        await driver.waitForSelector(globalMenuSettingsSelector);
+        await driver.clickElement(globalMenuSettingsSelector);
+
+        // Click Experimental tab
+        const securityAndPrivacyTabRawLocator = {
+          text: 'Experimental',
+          tag: 'div',
+        };
+        await driver.clickElement(securityAndPrivacyTabRawLocator);
+
+        // Toggle request queue setting
+        await driver.clickElement('.request-queue-toggle');
+
+        // Navigate to extension home screen
+        await driver.navigate(PAGES.HOME);
+
+        // Open Dapp One
+        await openDapp(driver, undefined, DAPP_URL);
+
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        // Network Selector
+        await driver.clickElement('[data-testid="network-display"]');
+
+        // Switch to second network
+        await driver.clickElement({
+          text: 'Localhost 8546',
+          css: 'p',
+        });
+
+        // TODO: Request Queuing bug when opening both dapps at the same time will have them stuck on the same network, with will be incorrect for one of them.
```

Comment:
```suggestion
        // TODO: Request Queuing bug when opening both dapps at the same time will have them stuck on the same network, which will be incorrect for one of them.
```

also, thats an odd one! defs something to look into. Might be worth making a branch that has a failing test as described here.